### ISSUE TITLE
Docker simulator not working, built in DB configuration problem

### ISSUE THREAD

**Issue by pmury** [apache/cloudstack#3397]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Issue with Docker based simulator
~~~

##### CLOUDSTACK VERSION

~~~
4.13 SNAPSHOT, "latest" image tag
~~~

##### CONFIGURATION

Docker CE 18.09.6 

##### OS / ENVIRONMENT

Ubuntu 18.04.2 LTS

##### SUMMARY

Can't access UI via localhost:8080: 503 error Service not available.
According to docker logs there's a database configuration problem:
Caused by: java.sql.SQLException: Access denied for user 'cloud'@'localhost' (using password: YES)

##### STEPS TO REPRODUCE

~~~
As README says: run Docker container and access it.
docker run --name simulator -p 8080:8080 -d cloudstack/simulator
~~~

##### EXPECTED RESULTS

~~~
Simulator UI is accessible
~~~

##### ACTUAL RESULTS

~~~
HTTP ERROR: 503

Problem accessing /client/. Reason:

    Service Unavailable
~~~

**Comment by pmury** [comment]
Seems like deployment scripts assume there should be a DB user named "cloud" but it is not in the database delivered with the simulator:
```

mysql> SELECT user, authentication_string FROM mysql.user;
+------------------+-------------------------------------------+
| user             | authentication_string                     |
+------------------+-------------------------------------------+
| root             |                                           |
| mysql.session    | *THISISNOTAVALIDPASSWORDTHATCANBEUSEDHERE |
| mysql.sys        | *THISISNOTAVALIDPASSWORDTHATCANBEUSEDHERE |
| debian-sys-maint | *2505A97210CB9BFDFA8B5C7453A7800A968ECAF7 |
+------------------+-------------------------------------------+
4 rows in set (0.00 sec)

```

**Comment by rhtyd** [comment]
Looks like fixed in https://github.com/apache/cloudstack/commit/9abb703a41a48fc16bc8153451b3c403d4522454
Please reopen if against latest master this is still reproducible

**Comment by martin-sweeny** [comment]
Experiencing this with docker simulator on `latest`

**Comment by tampler** [comment]
I'm having the same issue with the latest 4.15 code. Simulator seems to be totally unmainted on Docker as per [this](https://hub.docker.com/r/cloudstack/simulator).

After building the ACS I had to manually run `ssh-keygen` command to produce a `id_rsa.cloud` key and restart the simulator container. Now, I'm getting a null-pointer exception as below. The full log is here: [sim.log](https://github.com/apache/cloudstack/files/7702115/sim.log)
```bash
[0m INFO  [o.a.c.s.m.m.i.DefaultModuleDefinitionSet] (main:null) (logid:) Loading module context [root-ca] from URL [jar:file:/acs/client/target/cloud-client-ui-4.15.3.0-SNAPSHOT.jar!/META-INF/cloudstack/bootstrap/spring-bootstrap-context-inheritable.xml]
[0m INFO  [o.a.c.s.l.CloudStackExtendedLifeCycle] (main:null) (logid:) Configuring CloudStack Components
[0m [WARNING] Failed startup of context o.e.j.m.p.JettyWebAppContext@10f192d8{/client,file:///acs/client/target/classes/META-INF/webapp/,UNAVAILABLE}{file:///acs/client/target/classes/META-INF/webapp/}
[0m org.springframework.context.ApplicationContextException: Failed to start bean 'cloudStackLifeCycle'; nested exception is java.lang.NullPointerException
[0m     at org.springframework.context.support.DefaultLifecycleProcessor.doStart (DefaultLifecycleProcessor.java:181)
[0m     at org.springframework.context.support.DefaultLifecycleProcessor.access$200 (DefaultLifecycleProcessor.java:54)
[0m     at org.springframework.context.support.DefaultLifecycleProcessor$LifecycleGroup.start (DefaultLifecycleProcessor.java:356)
```

@rhtyd Could you pls take a look

**Comment by rhtyd** [comment]
Hi @tampler yes the docker support is not officially maintained, I think some of my colleagues may have some idea, cc @weizhouapache @DaanHoogland @sureshanaparti

**Comment by weizhouapache** [comment]
@tampler 
could you please test with 4.16.0.0 ?
issues should have been fixed by #5507

**Comment by tampler** [comment]
Hi @weizhouapache @rhtyd 

I've pulled Wei's image from [here](https://hub.docker.com/r/ustcweizhou/cloudstack-simulator). I was able to boot it. I'll be looking at it these days.

Wei, could you pls upload it to the official ACS account [here](https://hub.docker.com/r/cloudstack/simulator) And drop obsolete images not to confuse people.

**Comment by synergiator** [comment]
are here any updates? what could be blockers? the fix works!

**Comment by weizhouapache** [comment]
@synergiator 
I think this has been resolved by #5507 .

**Comment by synergiator** [comment]
@weizhouapache the fix makes possible creating your own container from the upstream codebase and this is great. However, the official images haven't been delivered yet where you'd expect them.

![image](https://user-images.githubusercontent.com/1834666/168592757-eb5c5d2b-2752-4972-880a-0440aad710ee.png)

Side note: the alternate Docker Hub image works, but only the basic DC deployment. Advanced configuration example scripts seem to fail.

![image](https://user-images.githubusercontent.com/1834666/168593221-21905a51-3f9c-470f-a2e1-1e3ff1b7c04a.png)
